pkg/cmd/render/render_test.go

```diff
@@ -488,3 +490,105 @@ type fakeBootstrapIPLocator struct {
 func (f *fakeBootstrapIPLocator) getBootstrapIP(ipv6 bool, machineCIDR string, excludedIPs []string) (net.IP, error) {
 	return f.ip, nil
 }
+
+const installConfigSingleStackIPv4 = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+const installConfigDualStack = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 2620:52:0:1302::/64
+  - cidr: 10.0.0.0/16
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+const installConfigSingleStackIPv6 = `
+apiVersion: v1
+metadata:
+  name: my-cluster
+networking:
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+  machineCIDR: 10.0.0.0/16
+  machineNetwork:
+  - foo: bar
+    cidr: 2620:52:0:1302::/64
+  networkType: OpenShiftSDN
+  serviceNetwork:
+  - 172.30.0.0/16
+`
+
+func Test_getMachineCIDR(t *testing.T) {
+	tests := map[string]struct {
+		installConfig     string
+		isSingleStackIPv6 bool
+		expectedCIDR      string
+		expectedErr       error
+	}{
+		"ipv4 should ignore the ipv6 cidr": {
+			installConfig:     installConfigSingleStackIPv4,
+			isSingleStackIPv6: false,
+			expectedCIDR:      "10.0.0.0/16",
+			expectedErr:       nil,
+		},
+		"dual stack should locate the ipv4 cidr": {
+			installConfig:     installConfigDualStack,
+			isSingleStackIPv6: false,
+			expectedCIDR:      "10.0.0.0/16",
+			expectedErr:       nil,
+		},
+		"single stack ipv6 should locate the ipv6 cidr": {
+			installConfig:     installConfigSingleStackIPv6,
+			isSingleStackIPv6: true,
+			expectedCIDR:      "2620:52:0:1302::/64",
+			expectedErr:       nil,
+		},
+	}
+
+	for name, test := range tests {
+		t.Logf("evaluating test %q", name)
```

Comment:
the cool modern way is to do

    t.Run(name, func(t *testing.T) {
            ...
    })

Reply:
Nice, thanks for the tip. Refactored